src/server/rdb_save.cc

```diff
@@ -854,27 +854,34 @@ void AppendFooter(io::StringSink* dump_res, bool ignore_crc) {
 }
 }  // namespace
 
-void SerializerBase::DumpObject(const CompactObj& obj, io::StringSink* out, bool ignore_crc) {
-  CompressionMode compression_mode = GetDefaultCompressionMode();
-  if (compression_mode != CompressionMode::NONE) {
-    compression_mode = CompressionMode::SINGLE_ENTRY;
+void SerializerBase::DumpObject(RdbSerializer* serializer, const CompactObj& obj,
+                                io::StringSink* out, bool ignore_crc) {
+  CompressionMode serializer_used_compression_mode = serializer->compression_mode_;
```

Comment:
Not sure should we save compression_mode and restore it later after we leave this function?

Reply:
this function is static I think